src/MockBot.Api/Services/ChatService.cs

```diff
@@ -0,0 +1,38 @@
+﻿using MockBot.Api.Interfaces;
+using MockBot.Api.Models;
+
+namespace MockBot.Api.Services;
+
+public class ChatService : IChatService
+{
+    private readonly Dictionary<Guid, Chat> _chats;
+
+    public ChatService()
+    {
+        _chats = new Dictionary<Guid, Chat>();
+    }
+
+    public Chat Create()
+    {
+        var chat = new Chat();
+        _chats.Add(chat.Id, chat);
+        return chat;
+    }
+
+    public IEnumerable<Chat> FindAll()
+    {
+        return _chats.Values.ToList();
+    }
+
+    public Chat Get(Guid id)
+    {
+        return _chats[id];
+    }
+
+    public Message CreateMessage(Guid chatId, string content)
```

Comment:
I'm not understanding why we have separated the `Create` and `CreateMessages` methods here? Would it be better to have `Create` take a `Chat` as a parameter which includes the `Messages`? Feel free to ping me and we can pair on it if you like?

Reply:
`Create` is to start a new chat, `CreateMessage` is to add a new message to an already existing chat.

Do you mean that you would rather see a single `Create` that would either create a new chat or a new message, based on whether it has any message data?

Also passing an entire `Chat` object with all its `Message` objects seems excessive just to add one new message.

Reply:
@joonasroosalung, OK that does make a bit more sense now... I suggest that these methods are renamed as `CreateChat` and `AddMessage`. This would make it clearer I think

Reply:
Of course